server/memphis_helper.go

```diff
@@ -361,19 +361,66 @@ func (s *Server) CreateConsumer(consumer models.Consumer, station models.Station
 		return err
 	}
 
-	err = s.memphisAddConsumer(stationName.Intern(), &ConsumerConfig{
-		Durable:       consumerName,
-		DeliverPolicy: DeliverAll,
-		AckPolicy:     AckExplicit,
-		AckWait:       time.Duration(maxAckTimeMs) * time.Millisecond,
-		MaxDeliver:    MaxMsgDeliveries,
-		FilterSubject: stationName.Intern() + ".final",
-		ReplayPolicy:  ReplayInstant,
-		MaxAckPending: -1,
-		HeadersOnly:   false,
-		// RateLimit: ,// Bits per sec
-		// Heartbeat: // time.Duration,
-	})
+	var ackPolicy DeliverPolicy
+	var consumerConfig *ConsumerConfig
+	streamInfo, err := serv.memphisStreamInfo(stationName.Intern())
+	if err != nil {
+		return errors.New("Streaminfo: " + err.Error())
+	}
+	lastSeq := streamInfo.State.LastSeq
+
+	if consumer.LastMessages != 0 {
+		lastMessages := (lastSeq - consumer.LastMessages) + 1
+		if int(lastMessages) < 0 {
+			lastMessages = uint64(1)
+		}
+		ackPolicy = DeliverByStartSequence
+		consumerConfig = &ConsumerConfig{
```

Comment:
you don't need to re-write all the keys in each block. write the keys appears in all blocks and inject those which can be changed